qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,677 @@
+            if len(depths) == 1:
+                # The BIOM table was rarefied
+                # Create the initial unrarefied artifact
+                initial_biom_artifact_id = create_non_rarefied_biom_artifact(
+                    analysis, biom_data, table)
+                # Create the rarefaction job
+                rarefaction_job_id, params = create_rarefaction_job(
+                    depths.pop(), initial_biom_artifact_id, analysis,
+                    srare_cmd_id)
+                # Create the rarefied artifact
+                rarefied_biom_artifact_id = create_rarefied_biom_artifact(
+                    analysis, srare_cmd_id, biom_data, params,
+                    initial_biom_artifact_id, rarefaction_job_id,
+                    srare_cmd_out_id)
+            else:
+                # The BIOM table was not rarefied, use current table as initial
+                initial_biom_id = transfer_file_to_artifact()
+
+            # Loop through all the jobs that used this biom table as input
+            sql = """SELECT *
+                     FROM qiita.job
+                     WHERE reverse(split_part(reverse(
+                        options::json->>'--otu_table_fp'), '/', 1)) = %s"""
+            TRN.add(sql, [filepath])
+            analysis_jobs = TRN.execute_fetchindex()
+            for job_data in analysis_jobs:
+                # Identify which command the current job exeucted
+                if job_data['command_id'] == 1:
+                    # Taxa summaries
+                    cmd_id = sum_taxa_cmd_id
+                    params = ('{"biom_table":%d,"metadata_category":"",'
+                              '"sort":false}' % initial_biom_id)
+                    output_artifact_type_id = ts_atype_id
+                    cmd_out_id = sum_taxa_cmd_out_id
+                elif job_data['command_id'] == 2:
+                    # Beta diversity
+                    cmd_id = bdiv_cmd_id
+                    tree_fp = loads(job_data['options'])['--tree_fp']
+                    params = ('{"biom_table":%d,"tree":"%s","metrics":'
+                              '["unweighted_unifrac","weighted_unifrac"]}'
+                              % (initial_biom_id, tree_fp))
+                    output_artifact_type_id = dm_atype_id
+                    cmd_out_id = bdiv_cmd_out_id
+                else:
+                    # Alpha rarefaction
+                    cmd_id = arare_cmd_id
+                    tree_fp = loads(job_data['options'])['--tree_fp']
+                    params = ('{"biom_table":%d,"tree":"%s","num_steps":"10",'
+                              '"min_rare_depth":"10",'
+                              '"max_rare_depth":"Default"}'
+                              % (initial_biom_id, tree_fp))
+                    output_artifact_type_id = rc_atype_id
+                    cmd_out_id = arare_cmd_out_id
+
+                transfer_job()
+
+with TRN:
+    # Unlink the analysis from the biom table filepaths
```

Comment:
Due to permissions in our DB, could you put this in a try/except and if it fails print the SQL commands?